### PR TITLE
Add flag for etag grouping

### PR DESCRIPTION
**Pull request by XiaofeiCao**
https://github.com/Azure/autorest.java/issues/2633

As a first step, add a flag for disabling etag headers grouping.

### CODE REVIEW

File: typespec-extension/src/code-model-builder.ts [hunk 1]
```diff
@@ -781,7 +785,9 @@ export class CodeModelBuilder {
     }
 
     // group ETag header parameters, if exists
-    this.processEtagHeaderParameters(codeModelOperation, op);
+    if (!this.options["disable-etag-grouping"]) {
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:788]: Just check `arm` would be enough for now?

Comment (XiaofeiCao) [reply typespec-extension/src/code-model-builder.ts:788]: For now it is. Srikanta mentioned that in the future we may want new SDKs enabled and existing ones disabled(avoid breaking). A flag would be helpful in this case. 

Agree a positive one sounds better. I'll go with `group-etag-headers`.

File: typespec-extension/src/code-model-builder.ts [hunk 2]
```diff
@@ -207,6 +207,8 @@ export class CodeModelBuilder {
       this.options["skip-special-headers"].forEach((it) => SPECIAL_HEADER_NAMES.add(it.toLowerCase()));
     }
 
+    this.options["group-etag-headers"] = this.options["group-etag-headers"] ?? true;
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:210]: We do not check `arm`?

File: typespec-extension/src/emitter.ts [hunk 3]
```diff
@@ -42,6 +42,7 @@ export interface EmitterOptions {
   "polling"?: any;
 
   "arm"?: boolean;
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/emitter.ts:44]: Not related to this PR.

`arm` is actually not a user settable option, right?
Then it better not be here at all.

If you need set it to Java, just do e.g. `(options as any).arm = codeModel.arm`

File: typespec-extension/src/code-model-builder.ts (hunk 2 above)
Comment (XiaofeiCao) [reply typespec-extension/src/code-model-builder.ts:210]: In the long term I think we don't. 
For green fields, before we fix etag headers grouping, we'll have to set this property to false in each tspconfig.yaml. After the fix, we can leave it out for green fields.
For brown fields, we'll always set it false.

Alternatively, as a short-term solution, we could do, if it's ARM and the flag is not set, we set it false. Actually, after a second thought, this one is quite appealing too, as we can selectively enable it for SDKs that really need it.

What's your opinion on this?

Comment (weidongxu-microsoft) [reply typespec-extension/src/code-model-builder.ts:210]: I am OK either way. Anyway, at present, if tsp has it, arm can only use false, so it just automatically be false, or you need to set it to false.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**